crud/select/plan.lua

```diff
@@ -106,10 +107,65 @@ local function extract_sharding_key_from_scan_value(scan_value, scan_index, shar
     return sharding_key
 end
 
+local function construct_after_tuple_by_fields(space_format, field_names, tuple)
+    if tuple == nil then
+        return nil
+    end
+
+    if field_names == nil then
+        return tuple
+    end
+
+    local positions = {}
+    local transformed_tuple = {}
+
+    for i, field in ipairs(space_format) do
+        positions[field.name] = i
+    end
+
+    for i, field_name in ipairs(field_names) do
+        local fieldno = positions[field_name]
+        if fieldno == nil then
+            return nil, FilterFieldsError:new(
+                    'Space format doesn\'t contain field named %q', field_name
+            )
+        end
+
+        transformed_tuple[fieldno] = tuple[i]
+    end
+
+    return transformed_tuple
+end
```

Comment:
Please, describe your logic in PR description. I don't understand why do you need to construct "after tuple".

Just pass it as the second argument from select.call_on_storage.

```lua
tuples, after_tuple = call()
if tuples == nil then
  -- after_tuple is error here
  return nil, after_tuple 
end

...

return {
  metadata = ...,
  rows = transform(tuples),
  after_tuple = after_tuple,
}
```

Anyway we will need to have some kind of cursor for each storage result when we will implement merger (https://github.com/tarantool/crud/pull/38/files#diff-3c146e04832d06f8726bb2eabde2e134695f8924f7847d3058d05ada80e837b5L114-R79)

Reply:
So you could really simplify reviewer's life if describe all transformations of single tuple: where, why.

E.g. "on storage we strip all fields except user defined fields and fields that are needed for comparison on router", "on router: ..."

Reply:
Well, ignore my first comment. But there is a lack of information about tuple lifecycle in case when "fields" is specified - so, the second comment is still relevant.